app/code/Magento/Quote/Model/QuoteAddressValidator.php

```diff
@@ -127,6 +127,6 @@ public function validate(AddressInterface $addressData)
      */
     public function validateForCart(CartInterface $cart, AddressInterface $address): void
     {
-        $this->doValidate($address, $cart->getCustomerIsGuest() ? null : $cart->getCustomer()->getId());
+        $this->doValidate($address, $cart->getCustomer()->getId() ? $cart->getCustomer()->getId() : null);
```

Comment:
Hello @sumitCueBlocks, thanks for your submission!
In this case, to reduce code redundancy, you can use the `?:` (Elvis) operator, thus the code would become:
`$this->doValidate($address, $cart->getCustomer()->getId() ?: null);
My best

Reply:
Hi @aleron75,

Thank you for your suggestion, I've updated the code as you suggested to reduce code redundancy.

Regards
Sumit